### PR TITLE
fix: Correctly prefill username in case of non-org email invite in an org

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- In case of non-org email(e.g. abc@gmail.com) when invited to an org, the signup page shows username as `abc` instead of `abc-gmail` - This issue is fixed now. It fixes the issue reported here https://github.com/calcom/cal.com/pull/12774#issuecomment-1860257694
- Also, username check wasn't considering the organization scope correctly and thus it was checking the username availability across all orgs - This is fixed now.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Tests (Unit/Integration/E2E or any other test)

## How should this be tested?
See https://github.com/calcom/cal.com/pull/12774#issuecomment-1860257694 for more details.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #12854** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12854?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12854?utm_source=stack-comment).

File: apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx [hunk 1]
```diff
@@ -74,7 +74,8 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
   const debouncedApiCall = useMemo(
     () =>
       debounce(async (username: string) => {
-        const { data } = await fetchUsername(username);
+        // TODO: Support orgSlug
+        const { data } = await fetchUsername(username, null);
```
Comment (hariombalhara) [inline apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx:78]: We should read the orgSlug here from the correct place and pass that on.

**Review by hariombalhara — COMMENTED** [review]
Self review done

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**